bin/watch.ts

```diff
@@ -20,26 +22,27 @@ function* start(): Sequence {
   });
 
   let current = { halt: (x = undefined) => x };
+  let context = yield self;
   let restart = () => {
     current.halt();
-    current = this.fork(function*() {
+    current = context.spawn(fork(function*() {
       try {
         yield launch(cmd, args);
         listener.halt();
       } catch (error) {
         console.log(error);
       }
-    })
+    }));
   };
 
   restart();
 }
 
-function* launch(cmd: string, args: string[]): Sequence {
+function* launch(cmd: string, args: string[]): Operation {
```

Comment:
`Sequence` is no longer an exported interface. Everything should be annotated as an `Operation`. This actually makes reasoning about composition a lot simpler.